TMEDIA 685 Pill

JackHowa opened the pull request:
## Ticket

- [TMEDIA-685](https://arcpublishing.atlassian.net/browse/TMEDIA-685)

## Description

Add pill component that can be used as link or text

## Acceptance Criteria

Provide a <Pill> component 

Can be a link or not

If it is a link, it should not use the link component → no nested components

Account for link states: Active, Hover, Visited, Focus

See [arc-themes-components/_index.scss at arc-themes-release-version-1.19 · WPMedia/arc-themes-components](https://github.com/WPMedia/arc-themes-components/blob/arc-themes-release-version-1.19/src/components/link/_index.scss#L6) for example of how to account for states. Do not use the link class though. Should be using c-pill class

Have Scss file that adds the hooks to the Sass library

Have test coverage

Have storybook documentation

Note: Until we have a theme switcher, there won’t be a background to see the styles

## Test Steps

1. Checkout branch - `git checkout TMEDIA-685-pill-component`
2. Update dependencies - `npm i`
3. Run Storybook `npm run storybook`
4. Check out the Pill storybook documentation
5. See classname used and passed in. 

## Author Checklist

- [x] Confirmed all the test steps a reviewer will follow above are working.
- [x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x] Confirmed relevant documentation has been updated/added.
- [x] Add label - **ready for review** when the pull request is ready for someone to begin reviewing

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] All GitHub Actions pass
- [ ] Ran the code locally based on the test instructions.
- [ ] Checked Chromatic for Storybook changes, accepted the updates if acceptable
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.
- [ ] Approve and Add label - **ready to merge** if you are happy with the pull request
- [ ] Want another reviewer? Add the label **additional review**

JackHowa committed:
feat: Add pill component along with description and examples

matthewroach commented on src/components/pill/_index.scss:
```diff
@@ -0,0 +1,17 @@
+		@include scss.component-properties("pill-active");
+	}
+
+	&:visited {
```
We are missing `:focus` hooks

JackHowa replied:
oops updated

JackHowa replied:
fixed as well for overline https://arcpublishing.atlassian.net/browse/TMEDIA-670

JackHowa committed:
feat: Add focus hook

JackHowa committed:
feat: Add overline focus as well similar requirements

Account for link states: Active, Hover, Visited, Focus  https://arcpublishing.atlassian.net/browse/TMEDIA-670
